### PR TITLE
[Muti backend] add heuristics_config for muti_bacnkend

### PR DESCRIPTION
**Pull request by Galaxy1458**
### PR Category
Other

### Type of Change
Other

### Description
Add FlagGems/src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py 
vendors need to config the HEURISTICS_CONFIGS
```python
HEURISTICS_CONFIGS = {
    "argmax": {
        "BLOCK_M": argmax_heur_block_m,
        "BLOCK_N": argmax_heur_block_n,
    },
   .....
```

### Progress

- [x] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### CODE REVIEW

File: src/flag_gems/runtime/__init__.py [hunk 1]
```diff
@@ -21,4 +21,8 @@ def get_triton_config(op_name):
     return config_loader.get_triton_config(op_name)
 
 
+def get_heuristics_config(op_name):
```
Comment (StrongSpoon) [inline src/flag_gems/runtime/__init__.py]: what about renaming them as get_tuned_config and get_heuristic_config?

File: src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    "index_select": {
+        "BLOCK_M": index_select_heur_block_m,
+        "BLOCK_N": index_select_heur_block_n,
+    },
+    "mm": {
+        "EVEN_K": mm_heur_even_k,
+    },
+    "multinomial": {
+        "NBLOCK": lambda args: 128,
+        "num_warps": lambda args: 4,
```
Comment (StrongSpoon) [inline src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py]: set the constant value as default

File: src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        "TILE_M": softmax_heru_tile_m,
+        "ONE_TILE_PER_CTA": softmax_heur_one_tile_per_cta,
+    },
+    "uniform": {
+        "BLOCK": uniform_heur_block,
+        "num_warps": uniform_heur_num_warps,
+    },
+    "upsample_nearest2d": {
+        "SAME_H": lambda args: args["OH"] == args["IH"],
+        "SAME_W": lambda args: args["OW"] == args["IW"],
```
Comment (StrongSpoon) [inline src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py]: I suggest defining these lambda expressions as functions.

File: src/flag_gems/runtime/__init__.py (hunk 1 above)
Comment (Galaxy1458) [reply src/flag_gems/runtime/__init__.py]: Done

File: src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py (hunk 2 above)
Comment (Galaxy1458) [reply src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py]: Done

File: src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py (hunk 3 above)
Comment (Galaxy1458) [reply src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py]: Done

File: src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py (hunk 2 above)
Comment (Galaxy1458) [reply src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py]: defining these lambda expressions as functions.

File: src/flag_gems/runtime/__init__.py [hunk 4]
```diff
@@ -17,8 +17,12 @@
...
+def get_tuned_config(op_name):
+    return config_loader.get_tuned_config(op_name)
 
 
-__all__ = ["commom_utils", "backend", "device", "get_triton_config"]
+def get_heuristic_config(op_name):
+    return config_loader.heuristics_config[op_name]
+
+
+__all__ = ["commom_utils", "backend", "device", "get_tuned_config"]
```
Comment (StrongSpoon) [inline src/flag_gems/runtime/__init__.py]: add get_heuristic_config

File: src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+
+def upsample_nearest2d_SAME_H(args):
+    return args["OH"] == args["IH"]
+
+
+def upsample_nearest2d_SAME_W(args):
+    return args["OW"] == args["IW"]
+
+
+def multinomial_NBLOCK(args):
```
Comment (StrongSpoon) [inline src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py]: what I mean is, we could set the values 128 and 4 as the default value of triton function parameters. modify the declaration code of multinomial kernel and there is no need to call heuristics.

Comment (Galaxy1458) [reply src/flag_gems/runtime/backend/_nvidia/heuristics_config_utils.py]: Done.  Although I don't know why the previous code was like this and the original plan was to keep the intention unchanged

File: src/flag_gems/runtime/__init__.py (hunk 4 above)
Comment (Galaxy1458) [reply src/flag_gems/runtime/__init__.py]: Done

File: src/flag_gems/ops/multinomial.py [hunk 6]
```diff
@@ -9,15 +9,9 @@
...
-@triton.heuristics(
-    {
-        "NBLOCK": lambda args: 128,
-        "num_warps": lambda args: 4,
-    }
-)
 @triton.jit(do_not_specialize=["K", "N", "philox_seed", "philox_offset"])
 def multinomial_with_replacement(
-    cdf_ptr, out_ptr, K, N, philox_seed, philox_offset, NBLOCK: tl.constexpr
+    cdf_ptr, out_ptr, K, N, philox_seed, philox_offset, NBLOCK: tl.constexpr = 128
```
Comment (StrongSpoon) [inline src/flag_gems/ops/multinomial.py:14]: yes, that's it.

**Review by StrongSpoon — APPROVED** [review]
lgtm

**Pull request merged**